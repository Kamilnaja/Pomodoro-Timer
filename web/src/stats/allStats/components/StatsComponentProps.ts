import { StatsSearchResult } from '../../../../../types/statisticsInterfaces';
import { HandleGetStats } from '../../store/models/StatsInterfaces';

export type StatsComponentProps = { stats: StatsSearchResult; handleGetStats: HandleGetStats };
