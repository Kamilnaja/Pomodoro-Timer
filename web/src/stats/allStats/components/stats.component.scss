@import 'shared/styles/shared.scss';

.stats {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 0 auto;
  max-width: $page-width;
  padding: 20px 0;
  text-align: center;

  & &__today-result {
    margin-bottom: 20px;
  }
}

.navigation {
  align-items: flex-end;
  display: flex;
  flex-direction: row;
  margin: 0 auto;
  min-width: 200px;
}
