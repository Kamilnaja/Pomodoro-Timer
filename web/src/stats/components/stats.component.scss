@import "./../../shared/styles/shared.scss";

.stats {
  @include font-fira-sans($size: 15px, $colour: $text-secondary);
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 0 auto;
  max-width: $page-width;
  padding: 20px 0;
  text-align: center;

  & &__today-result {
    margin-bottom: 20px;
  }
}

.table {
  border: white;
  margin: 0 auto;
  max-width: 600px;
  width: 100%;

  & &__head {
    border-bottom: 1px dashed white;
  }
}
