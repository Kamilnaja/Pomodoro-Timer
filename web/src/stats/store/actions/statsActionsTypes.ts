import { StatsSearchResult } from '../../../../../types/statisticsInterfaces';

export const GET_STATISTIC_IN_PERIOD = 'GET_STATISTIC_IN_PERIOD';
export const GET_STATISTIC_IN_PERIOD_SUCCESS = 'GET_STATISTIC_IN_PERIOD_SUCCESS';
export const GET_STATISTIC_IN_PERIOD_ERROR = 'GET_STATISTIC_IN_PERIOD_ERROR';

export const GET_TODAY_STATS = 'GET_TODAY_STATS';
export const GET_TODAY_STATS_SUCCESS = 'GET_TODAY_STATS_SUCCESS';
export const GET_TODAY_STATS_ERROR = 'GET_TODAY_STATS_ERROR';

export const SAVE_POMODORO = 'SAVE_POMODORO';
export const SAVE_POMODORO_SUCCESS = 'SAVE_POMODORO_SUCCESS';
export const SAVE_POMODORO_ERROR = 'SAVE_POMODORO_ERROR';

export interface GetStatisticsInPeriodAction {
  type: typeof GET_STATISTIC_IN_PERIOD;
}

export interface GetStatisticsInPeriodSuccessAction {
  type: typeof GET_STATISTIC_IN_PERIOD_SUCCESS;
  payload: StatsSearchResult;
}

export interface GetStatisticsInPeriodErrorAction {
  type: typeof GET_STATISTIC_IN_PERIOD_ERROR;
  payload: any;
}

export interface GetTodayStatsAction {
  type: typeof GET_TODAY_STATS;
}

export interface GetTodayStatsSuccessAction {
  type: typeof GET_TODAY_STATS_SUCCESS;
  payload: number;
}

export interface GetTodayStatsErrorAction {
  type: typeof GET_TODAY_STATS_ERROR;
  payload: any;
}

export interface SavePomodoroAction {
  type: typeof SAVE_POMODORO;
}

export interface SavePomodoroSuccessAction {
  type: typeof SAVE_POMODORO_SUCCESS;
}

export interface SavePomodoroErrorAction {
  type: typeof SAVE_POMODORO_ERROR;
  payload: any;
}

export type StatsActionsTypes =
  | GetStatisticsInPeriodAction
  | GetStatisticsInPeriodSuccessAction
  | GetStatisticsInPeriodErrorAction
  | GetTodayStatsAction
  | GetTodayStatsSuccessAction
  | GetTodayStatsErrorAction
  | SavePomodoroAction
  | SavePomodoroSuccessAction
  | SavePomodoroErrorAction;
