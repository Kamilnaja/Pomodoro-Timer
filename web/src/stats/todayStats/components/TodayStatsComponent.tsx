import React from 'react';
import { Badge } from 'react-bootstrap';
import { SearchResult } from '../../../../../types/commonInterfaces';

export const TodayStatsComponent = (props: SearchResult<number>) => {
  return (
    <p className="text-white text-center">
      Today you have made: <Badge>{props.result}</Badge> pomodoros
    </p>
  );
};
