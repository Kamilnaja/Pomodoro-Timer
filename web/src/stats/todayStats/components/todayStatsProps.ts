export interface TodayStatsProps {
  result: number;
}
