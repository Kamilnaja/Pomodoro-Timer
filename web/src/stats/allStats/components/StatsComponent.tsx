import { Button, ButtonGroup, Table } from 'react-bootstrap';
import { getCurrentMonth, getCurrentYear } from '../../../shared/scripts/utils';
import './stats.component.scss';
import { StatsComponentProps } from './StatsComponentProps';

let pageMonth = getCurrentMonth();
let pageYear = getCurrentYear();

export const StatsComponent = (props: StatsComponentProps) => {
  const getPreviousMonth = () => {
    if (pageMonth === 0) {
      pageMonth = 11;
      pageYear = pageYear - 1;
    } else {
      pageMonth = pageMonth - 1;
    }

    props.handleGetStats(pageYear, pageMonth);
  };

  const getNextMonth = (): void => {
    if (pageMonth === 11) {
      pageMonth = 0;
      pageYear = pageYear + 1;
    } else {
      pageMonth = pageMonth + 1;
    }

    props.handleGetStats(pageYear, pageMonth);
  };

  // const renderLineChart = (
  //   <LineChart width={600} height={300} data={props.stats}>
  //     <Line type="monotone" dataKey="count" stroke="#8884d8" />
  //   </LineChart>
  // );

  return (
    <div className="stats">
      {/* {renderLineChart} */}
      <ButtonGroup className="stats__navigation navigation">
        <Button
          className="navigation__button"
          onClick={() => getPreviousMonth()}
          disabled={!props.stats.hasPreviousPeriod}
        >
          &lt; prev
        </Button>
        <Button className="navigation__button" onClick={() => getNextMonth()} disabled={!props.stats.hasNextPeriod}>
          next &gt;
        </Button>
      </ButtonGroup>
      <Table className="stats__table table">
        <caption>
          Results from: {pageMonth + 1}.{pageYear}
        </caption>
        <thead className="table__head">
          <tr>
            <th>Date</th>
            <th>Number of pomodoros</th>
          </tr>
        </thead>
        <tbody>
          {props.stats.pomodoros?.map((v, idx) => (
            <tr key={idx}>
              <td className="table__date">{v.date}</td>
              <td className="table__count">{v.count}</td>
            </tr>
          ))}
        </tbody>
      </Table>
    </div>
  );
};
