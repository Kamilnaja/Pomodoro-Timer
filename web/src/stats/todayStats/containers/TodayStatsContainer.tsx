import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getCurrentMonth, getCurrentYear } from '../../../shared/scripts/utils';
import { StatsState } from '../../containers/statsContainerInterfaces';
import { TodayStatsComponent } from '../components/TodayStatsComponent';
import { handleGetTodayStats } from './../../store/actions/statsActions';
import { TodayStatsContainerProps } from './todayStatsContainerProps';

class TodayStatsContainer extends Component<TodayStatsContainerProps> {
  componentDidMount() {
    this.props.handleGetTodayStats({year: getCurrentYear(), month: getCurrentMonth(), day: });
  }

  render() {
    return <TodayStatsComponent result={this.props.todayStats} />;
  }
}

const mapStateToProps = (state: StatsState) => {
  const todayStats = state.todayResults;
  return { todayStats };
};

const mapDispatchToProps = {
  handleGetTodayStats,
};

export default connect(mapStateToProps, mapDispatchToProps)(TodayStatsContainer);
