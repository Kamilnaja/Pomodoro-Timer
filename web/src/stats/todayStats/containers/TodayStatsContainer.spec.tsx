import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { Store } from 'redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { StatsActionsTypes } from '../../store/actions/statsActionsTypes';
import { createStatsState } from '../../testing/stats.test.data';
import { TodayStatsComponent } from '../components/TodayStatsComponent';
import TodayStatsContainer from './TodayStatsContainer';

const mockStore = configureStore([thunk]);

describe('StatsContainer', () => {
  let component: renderer.ReactTestRenderer;
  let store: Store<any, StatsActionsTypes>;

  beforeEach(() => {
    store = mockStore({
      stats: createStatsState(),
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <TodayStatsContainer />
      </Provider>,
    );
  });

  it('should match snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
