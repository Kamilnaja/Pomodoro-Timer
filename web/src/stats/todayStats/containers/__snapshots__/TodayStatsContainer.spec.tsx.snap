// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsContainer should match snapshot 1`] = `
<p
  className="text-white text-center mt-2"
>
  Today you have made: 
  <span
    className="badge badge-primary"
  >
    10
  </span>
   pomodoros
</p>
`;
